toolchain/lex/tokenized_buffer.h

```diff
@@ -260,32 +260,93 @@ class TokenizedBuffer : public Printable<TokenizedBuffer> {
     int indent;
   };
 
-  struct TokenInfo {
-    TokenKind kind;
+  class TokenInfo {
+   public:
+    auto kind() const -> TokenKind { return TokenKind::FromInt(kind_); }
+    auto has_leading_space() const -> bool { return has_leading_space_; }
+
+    auto ident_id() const -> IdentifierId {
+      CARBON_DCHECK(kind() == TokenKind::Identifier);
+      return IdentifierId(token_payload_);
+    }
+    auto set_ident_id(IdentifierId ident_id) -> void {
+      CARBON_DCHECK(kind() == TokenKind::Identifier);
+      CARBON_DCHECK(ident_id.index < (2 << PayloadBits));
+      token_payload_ = ident_id.index;
+    }
+
+    auto string_literal_id() const -> StringLiteralValueId {
+      CARBON_DCHECK(kind() == TokenKind::StringLiteral);
+      return StringLiteralValueId(token_payload_);
+    }
+
+    auto int_id() const -> IntId {
+      CARBON_DCHECK(kind() == TokenKind::IntLiteral ||
+                    kind() == TokenKind::IntTypeLiteral ||
+                    kind() == TokenKind::UnsignedIntTypeLiteral ||
+                    kind() == TokenKind::FloatTypeLiteral);
+      return IntId(token_payload_);
+    }
+
+    auto real_id() const -> RealId {
+      CARBON_DCHECK(kind() == TokenKind::RealLiteral);
+      return RealId(token_payload_);
+    }
+
+    auto closing_token_index() const -> TokenIndex {
+      CARBON_DCHECK(kind().is_opening_symbol());
+      return TokenIndex(token_payload_);
+    }
+    auto set_closing_token_index(TokenIndex closing_index) -> void {
+      CARBON_DCHECK(kind().is_opening_symbol());
+      CARBON_DCHECK(closing_index.index < (2 << PayloadBits));
+      token_payload_ = closing_index.index;
+    }
+
+    auto opening_token_index() const -> TokenIndex {
+      CARBON_DCHECK(kind().is_closing_symbol());
+      return TokenIndex(token_payload_);
+    }
+    auto set_opening_token_index(TokenIndex opening_index) -> void {
+      CARBON_DCHECK(kind().is_closing_symbol());
+      CARBON_DCHECK(opening_index.index < (2 << PayloadBits));
+      token_payload_ = opening_index.index;
+    }
+
+    auto error_length() const -> int {
```

Comment:
Is the `int` deliberate?

```suggestion
    auto error_length() const -> int32_t {
```

Reply:
This isn't stored once we extract it, so was trying to revert to non-bit-sized types.